app/models/attachment_visibility.rb

```diff
@@ -37,83 +37,30 @@ def initialize(attachment_data, user)
   end
 
   def visible?
-    visible_edition? || visible_policy_group? || visible_consultation_response?
+    attachment_data.visible_to?(user)
   end
 
   def unpublished_edition
-    attachable_ids = edition_ids + consultation_ids
-    if (unpublishing = Unpublishing.where(edition_id: attachable_ids).first)
-      Edition.find_by(id: unpublishing.edition_id)
-    end
+    attachment_data.unpublished_edition
   end
 
   def visible_attachment
-    if visible_attachable
-      (visible_attachable.attachments & attachment_data.attachments).first
-    end
+    attachment_data.visible_attachment_for(user)
   end
 
   def visible_attachable
-    visible_edition || visible_consultation_response || visible_policy_group
+    attachment_data.visible_attachable_for(user)
   end
 
   def visible_edition
-    visible_edition_scope.last
+    attachment_data.visible_edition_for(user)
   end
 
   def visible_consultation_response
-    if visible_consultation_response?
-      Response.where(edition_id: consultation_ids).last
-    end
+    attachment_data.visible_attachable_for(user)
```

Comment:
We discussed explaining why this is OK in the commit note, i.e. why is it OK that `#visible_attachable_for` returns a `PolicyGroup` or `Edition` here?

Reply:
I've added the following paragraph to the commit note:

> Note that `AttachmentVisibility#visible_consultation_response` is only called directly from `AttachmentVisibilityTest` and so, as a bit of a cheat, I'm calling `AttachmentData#visible_attachable_for`, even though it could conceivably return something other than a consultation response, e.g. a policy group or an edition. We should be able to remove `AttachmentVisibility#visible_consultation_response` altogether in the
near future.

Reply:
It turns out I forgot to push up the changes to the commit note before I merged the PR, so the above comment will have to suffice - sorry!